app/src/main/kotlin/com/wire/android/ui/home/conversations/media/ConversationMediaScreen.kt

```diff
@@ -69,31 +92,96 @@ fun ConversationMediaScreen(navigator: Navigator) {
     continueAssetLoading: (shouldContinue: Boolean) -> Unit
 ) {
+    val scope = rememberCoroutineScope()
+    val lazyListStates: List<LazyListState> = ConversationMediaScreenTabItem.entries.map { rememberLazyListState() }
+    val initialPageIndex = ConversationMediaScreenTabItem.PICTURES.ordinal
+    val pagerState = rememberPagerState(initialPage = initialPageIndex, pageCount = { ConversationMediaScreenTabItem.entries.size })
+    val maxAppBarElevation = MaterialTheme.wireDimensions.topBarShadowElevation
+    val currentTabState by remember { derivedStateOf { pagerState.calculateCurrentTab() } }
+    val elevationState by remember { derivedStateOf { lazyListStates[currentTabState].topBarElevation(maxAppBarElevation) } }
+
     WireScaffold(
         modifier = Modifier
             .background(color = colorsScheme().backgroundVariant),
         topBar = {
             WireCenterAlignedTopAppBar(
-                elevation = dimensions().spacing0x,
+                elevation = elevationState,
                 title = stringResource(id = R.string.label_conversation_media),
                 navigationIconType = NavigationIconType.Back,
-                onNavigationPressed = onNavigationPressed
+                onNavigationPressed = onNavigationPressed,
+                bottomContent = {
+                    WireTabRow(
+                        tabs = ConversationMediaScreenTabItem.entries,
+                        selectedTabIndex = currentTabState,
+                        onTabChange = { scope.launch { pagerState.animateScrollToPage(it) } }
+                    )
+                }
             )
         },
     ) { padding ->
-        // TODO implement tab here for https://wearezeta.atlassian.net/browse/WPB-5378
+        var focusedTabIndex: Int by remember { mutableStateOf(initialPageIndex) }
+
+        CompositionLocalProvider(LocalOverscrollConfiguration provides null) {
+            HorizontalPager(
+                state = pagerState,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(padding)
+            ) { pageIndex ->
+                when (ConversationMediaScreenTabItem.entries[pageIndex]) {
+                    ConversationMediaScreenTabItem.PICTURES -> PicturesContent(
+                        uiAssetMessageList = state.messages,
+                        onImageFullScreenMode = onImageFullScreenMode,
+                        continueAssetLoading = continueAssetLoading
+                    )
+                    ConversationMediaScreenTabItem.FILES -> FilesContent()
+                }
+            }
+
+            LaunchedEffect(pagerState.isScrollInProgress, focusedTabIndex, pagerState.currentPage) {
+                if (!pagerState.isScrollInProgress && focusedTabIndex != pagerState.currentPage) {
+                    focusedTabIndex = pagerState.currentPage
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun PicturesContent(
```

Comment:
Suggestion: we can add a preview for this :D so we are covering our backs for when screenshot testing comes into place.